### ISSUE TITLE
Error code 433

### ISSUE THREAD

**Issue by SLornieCYC** [data-to-insight/csc-validator-be-903#500]
Current implementation returns too many false-positives:

```
            episodes['original_index'] = episodes.index
            episodes.sort_values(['CHILD', 'DECOM', 'DEC'], inplace=True)
            episodes[['PREVIOUS_DEC', 'PREVIOUS_CHILD']] = episodes[['DEC', 'CHILD']].shift(1)

            rne_is_ongoing = episodes['RNE'].str.upper().astype(str).isin(['P', 'L', 'T', 'U', 'B'])
            date_mismatch = episodes['PREVIOUS_DEC'] != episodes['DECOM']
            missing_date = episodes['PREVIOUS_DEC'].isna() | episodes['DECOM'].isna()
            same_child = episodes['PREVIOUS_CHILD'] == episodes['CHILD']

            error_mask = rne_is_ongoing & (date_mismatch | missing_date) & same_child
```

Any child with an open episode at `<COLLECTION_START_DATE>` will flag as an error unless the open episode (which started in the previous year) happens to have `<RNE>` = S.  This is because the previous episode, being in the previous year, is not available to the `.shift` function and later tests.

Suggest this rule could be modified for the tool to :

- Only test episodes where the `<DECOM>` is also on or after `<COLLECTION_START_DATE>`

**Comment by dezog** [comment]
thanks for this, changed it to:
 ```
not_from_prev_yr = episodes['DECOM_dt'] >= collection_start
error_mask = rne_is_ongoing & (date_mismatch | missing_date) & same_child & not_from_prev_yr
```

**Comment by dezog** [comment]
I would have thought the `same_child` constraint would have prevented the first episode being counted as an error anyway... I'll tell you when this change has gone through to the actual site, and please let me know if it's sorted the issue

**Comment by dezog** [comment]
It was sorting on the dates as strings, not datetimes! I think that should fix it

**Comment by dezog** [comment]
Should be fixed by 20f446e1f639ef71c573fb2f1f258fe68e010ee9
